framework/src/org/checkerframework/common/value/Range.java

```diff
@@ -0,0 +1,670 @@
+     *         // cannot be greater than or equal to variable <i>b</i> with range [3, 7].
+     *         ...
+     *     }
+     * </code>
+     * </pre>
+     *
+     * @param right the specified {@code Range} to compare with
+     * @return the refined {@code Range}
+     */
+    public Range refineGreaterThanEq(Range right) {
+        if (this.isNothing() || right.isNothing()) {
+            return NOTHING;
+        }
+
+        long resultFrom = Math.max(from, right.from);
+        return createRangeOrNothing(resultFrom, to);
+    }
+
+    /**
+     * Refines this range to reflect that some value in it can be equal to a value in the given
+     * range. This is used for calculating the control-flow-refined result of the == operator. For
+     * example:
+     *
+     * <pre>
+     * <code>
+     *     {@literal @}IntRange(from = 0, to = 10) int a;
+     *     {@literal @}IntRange(from = 3, to = 15) int b;
+     *     ...
+     *     if (a == b) {
+     *         // range of <i>a</i> is now refined to [3, 10] because a value in range [0, 2]
+     *         // cannot be equal to variable <i>b</i> with range [3, 15].
+     *         ...
+     *     }
+     * </code>
+     * </pre>
+     *
+     * @param right the specified {@code Range} to compare with
+     * @return the refined {@code Range}
+     */
+    public Range refineEqualTo(Range right) {
+        if (this.isNothing() || right.isNothing()) {
+            return NOTHING;
+        }
+
+        long resultFrom = Math.max(from, right.from);
+        long resultTo = Math.min(to, right.to);
+        return createRangeOrNothing(resultFrom, resultTo);
+    }
+
+    /**
+     * Returns the number of possible values enclosed by this range. To prevent overflow, we use
+     * BigInteger for calculation and return a BigInteger.
+     *
+     * @return the number of possible values enclosed by this range
+     */
+    public BigInteger numberOfPossibleValues() {
+        return BigInteger.valueOf(to).subtract(BigInteger.valueOf(from)).add(BigInteger.valueOf(1));
+    }
+
+    /**
+     * Determines if the range is wider than a given value, i.e., if the number of possible values
+     * enclosed by this range is more than the given value.
+     *
+     * @param value the value to compare with
+     * @return true if wider than the given value
+     */
+    public boolean isWiderThan(long value) {
+        return numberOfPossibleValues().compareTo(BigInteger.valueOf(value)) == 1;
+    }
+
+    /**
+     * Determines if this range is completely contained in the scope of the Integer type.
+     *
+     * @return true if the range is completely contained in the scope of the Integer type
+     */
+    public boolean isWithinInteger() {
+        return from >= Integer.MIN_VALUE && to <= Integer.MAX_VALUE;
+    }
+
+    private static final BigInteger longPossibleValues =
+            BigInteger.valueOf(Long.MAX_VALUE - Long.MIN_VALUE + 1);
```

Comment:
I extracted this computation into a variable so it is not computed over and over, which would be inefficient.
However, it is incorrect.  It suffers overflow in the Long computation that is the argument to `BigInteger.valueOf`.
Can you fix this and write some test cases for routines that are currently untested?  (This will also be a motivation to remove unused routines -- so you do not have to test them.)